Description: Fix compilation of slamr module with 2.6.36
  ioctl got deprecated since kernel 2.6.36
Origin: http://bugs.gentoo.org/attachment.cgi?id=259296
Bug-Gentoo: http://bugs.gentoo.org/show_bug.cgi?id=302456
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=613906
Bug-Ubuntu: https://bugs.launchpad.net/bugs/737650
--- a/drivers/amrmo_init.c	2011-01-06 20:12:59.000000000 +0100
+++ b/drivers/amrmo_init.c	2011-01-06 20:12:34.000000000 +0100
@@ -437,11 +437,11 @@ static unsigned int amrmo_poll(struct fi



-static int amrmo_ioctl(struct inode *inode, struct file *file, unsigned int cmd, unsigned long parg)
+static long amrmo_unlocked_ioctl(struct file *file, unsigned int cmd, unsigned long parg)
 {
	struct amrmo_struct *amrmo = (struct amrmo_struct *)file->private_data;
	unsigned long flags;
-	int ret;
+	long ret;
	unsigned stat;
	unsigned long arg=0;

@@ -514,7 +514,7 @@ static struct file_operations amrmo_fops
	.read =    amrmo_read,
	.write =   amrmo_write,
	.poll =    amrmo_poll,
-	.ioctl =   amrmo_ioctl,
+	.unlocked_ioctl =   amrmo_unlocked_ioctl,
	.open =    amrmo_open,
	.release = amrmo_release,
 };
